components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPTransportSender.java

```diff
@@ -214,32 +219,30 @@ public void sendMessage(MessageContext messageContext, String targetEPR,
                     }
                     processResponse(messageContext, responseXML);
                 } catch (Exception e) {
+                    log.error("Error while sending request", e);
                     if (isTransaction(messageContext)) {
-                        //if something went wrong during the transaction, then rollback
+                        //Rollback transaction if something goes wrong during the transaction
                         JCoFunction rollbackFunction = getRFCfunction(destination,
                                 SAPConstants.BAPI_TRANSACTION_ROLLBACK);
                         evaluateRFCfunction(rollbackFunction, destination, escapeErrorHandling);
-                        if (log.isDebugEnabled()){
-                            log.debug("Rollback transaction.");
-                        }
+                        log.warn("Rolled-back transaction.");
                     }
                     sendFault(messageContext, e , SAP_TRANSPORT_ERROR);
                 } finally {
                     if (isTransaction(messageContext) || isLogon) {
                         //end transaction
                         JCoContext.end(destination);
-                        if (log.isDebugEnabled()){
-                            log.debug("End transaction.");
-                        }
+                        log.info("Ended transaction.");
```

Comment:
Should this be an info log? It feels like a debug log to me.

Reply:
done